src/mongocxx/include/mongocxx/v1/indexes.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/v1/indexes-fwd.hpp>
+
+//
+
+#include <mongocxx/v1/detail/prelude.hpp>
+
+namespace mongocxx {
+namespace v1 {
+
+///
+/// Support for MongoDB indexes.
+///
+/// @note Not to be confused with Atlas Search indexes or Atlas Vector Search indexes.
```

Comment:
The distinction between `index` and `index_view` does not seem valuable (perhaps it might have been when [CXX-1360](https://jira.mongodb.org/browse/CXX-1360) was resolved in Jun 2017...?). Besides the unnecessary "view" vs. "value" terminology (a la bsoncxx API), it may also be confused with [MongoDB standard or on-demand materialized views](https://www.mongodb.com/docs/manual/core/views/) as pertains to aggregation pipelines, which this class is _not_ directly related to:

> Standard views use the indexes of the underlying collection. As a result, you cannot create, drop or re-build indexes on a standard view directly, nor get a list of indexes on the view. You can create indexes directly on on-demand materialized views because they are stored on disk.